core/build.gradle

```diff
@@ -1,11 +1,13 @@
 description = 'gRPC: Core'
 
+
 dependencies {
     compile project(':grpc-context'),
             libraries.gson,
             libraries.guava,
             libraries.errorprone,
-            libraries.jsr305
+            libraries.jsr305,
+            libraries.animalsniffer_annotations
```

Comment:
compileOnly. The annotation is Retention(Class)

Reply:
done

Reply:
compileOnly triggers a Proguard warning (=> fails the Android build):

```
Warning: io.grpc.internal.JndiResourceResolverFactory$JndiResourceResolver: can't find referenced class org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
```

I think JndiResourceResolverFactory should actually be stripped out by proguard, so adding a proguard -dontwarn is probably all that's necessary - but this would then be required in all downstream apps' proguard configs as well. Will look for a better solution here...

Reply:
I see a few options here: (fwiw, guava had some similar pain tracked in https://github.com/google/guava/issues/2721)

1) Keep the dependency as compileOnly, which will require Android apps to add a new -dontwarn to their proguard configs
2) Bite the cost of making this a compile dependency in open-source. The annotation package is very small (no impact on Android app size/dex count).
3) Use our own custom annotation (supported by animalsniffer) to avoid the dependency
4) Drop the annotations altogether and configure animalsniffer to ignore JndiResourceResolverFactory's uses of unsupported APIs

I lean towards option 4. @ejona86 what do you think?

Reply:
(2) sounds appropriate. Let's just add it as a normal dep. Unfortunate, but whatever.